x-pack/plugins/security_solution/common/detection_engine/schemas/common/schemas.ts

```diff
@@ -111,6 +111,12 @@ export type IdOrUndefined = t.TypeOf<typeof idOrUndefined>;
 export const index = t.array(t.string);
 export type Index = t.TypeOf<typeof index>;
 
+export const data_view_id = t.string;
+export type DataViewId = t.TypeOf<typeof data_view_id>;
+
+export const dataViewIdOrUndefined = t.union([data_view_id, t.undefined]);
+export type DataViewIdOrUndefined = t.TypeOf<typeof dataViewIdOrUndefined>;
```

Comment:
Looks like we may phasing out the `OrUndefined` pattern presently in place -- more details on that over here in this comment https://github.com/elastic/kibana/pull/132409#discussion_r878711020. No change needed, but just a heads up! 🙂 

---
Also wanted to point out that @banderror mentioned over in https://github.com/elastic/kibana/pull/132409 that if the `const` declaration is the same name as the type the JSDoc's will transfer over as well, which is a nice little usability perk in the IDE. You can see an example over in the new rule_params:

https://github.com/elastic/kibana/blob/fb1eeb0945e25928a2516133055f048ff098166f/x-pack/plugins/security_solution/common/detection_engine/schemas/common/rule_params.ts#L46-L57

Reply:
This is neat! Haven't seen this before. I think it would be cool to come through and update the naming conventions to match this. I don't quite want to deviate from the naming conventions right in this moment but I think knowing there are benefits to changing it is great motivation for tech-debt cleanup down the line.